Core/Core/Assignments/AssignmentDetails/AssignmentSubmissionBreakdownViewModel.swift

```diff
@@ -0,0 +1,83 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2022-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public class AssignmentSubmissionBreakdownViewModel: SubmissionBreakdownViewModelProtocol {
+
+    @Published public var isReady: Bool = false
+    @Published public var graded: Int = 0
+    @Published public var ungraded: Int = 0
+    @Published public var unsubmitted: Int = 0
+    @Published public var submissionCount: Int = 0
+    @Published public var showError: Bool = false
+    @Published public var errorText: String?
+
+    public var noSubmissionTypes: Bool {
+        submissionTypes.contains(.not_graded) || submissionTypes.contains(.none)
+    }
+
+    public var paperSubmissionTypes: Bool {
+        submissionTypes.contains(.on_paper)
+    }
+
+    private let assignmentID: String
+    private let courseID: String
+    private let submissionTypes: [SubmissionType]
+    private var summary: Store<GetSubmissionSummary>
+    private var submissionsPath: String { "courses/\(courseID)/assignments/\(assignmentID)/submissions" }
+
+    init(courseID: String, assignmentID: String, submissionTypes: [SubmissionType]) {
+        self.assignmentID = assignmentID
+        self.courseID = courseID
+        self.submissionTypes = submissionTypes
+        summary = AppEnvironment.shared.subscribe(GetSubmissionSummary(
+            context: .course(courseID),
+            assignmentID: assignmentID
+        ))
+    }
+
+    public func viewDidAppear() {
+        summary.eventHandler = update
```

Comment:
This eventHandler will create a retain cycle. The view model strongly references the store and the store strongly references the view model to be able to call its `update` method.
```suggestion
        summary.eventHandler = { [weak self] in
            self?.update()
        }
```